lib/msf/core/exploit/remote/smb/client/microsoft_kerberos_authenticator.rb

```diff
@@ -0,0 +1,196 @@
+# -*- coding: binary -*-
+
+#
+# This class acts as standalone authenticator for Microsoft Kerberos and SMB
+#
+
+class Msf::Exploit::Remote::SMB::Client::MicrosoftKerberosAuthenticator
```

Comment:
So while working on Kerberos authentication for LDAP, I noticed that this class works as-is. Even with the service name being `cifs`, though that value should probably be configurable.

Would it be possible to move this code to a namespace that reflects that it is not SMB-client specific?

Reply:
I've moved this, and added a base class - now specific implementations which can override the principal name etc as required